### PR TITLE
Update topics/grains doco, about considerations before adding a Grain

### PR DESCRIPTION
**Pull request by damon-atkins**
### What does this PR do?

topics/grains regarding consideration to take into effect before adding a grain.
### What issues does this PR fix or reference?

Helps people understand when a grain should be created and what to place in it.
### Previous Behavior

Not Clear
### New Behavior

A bit clearer

### CODE REVIEW

File: doc/topics/grains/index.rst [hunk 1]
```diff
@@ -132,6 +132,15 @@ Using Jinja templating, only one match entry needs to be defined.
 Writing Grains
 ==============
 
+Consider adding Gains when the information is needed for targeting,
```
Comment (terminalmage) [inline doc/topics/grains/index.rst]: Misspelling and unnecessary capitalization. Also this sentence needs to be improved, or split into two sentences. As one sentence, it doesn't make much sense.

File: doc/topics/grains/index.rst [hunk 2]
```diff
@@ -132,6 +132,15 @@ Using Jinja templating, only one match entry needs to be defined.
...
+the grains should only contain the **bare minimum data** required. Also think
+about other platforms/operating systems, and if the grain key name and data
+structure can be named and design to support many platforms, operating systems
+or applications. Most of the time execution module called from a JINJA template
```
Comment (terminalmage) [inline doc/topics/grains/index.rst]: "jinja" should not be all capitals.

File: doc/topics/grains/index.rst [hunk 3]
```diff
@@ -132,6 +132,15 @@ Using Jinja templating, only one match entry needs to be defined.
 Writing Grains
 ==============
 
+Consider adding Gains when the information is needed for targeting,
+the grains should only contain the **bare minimum data** required. Also think
+about other platforms/operating systems, and if the grain key name and data
+structure can be named and design to support many platforms, operating systems
```
Comment (terminalmage) [inline doc/topics/grains/index.rst]: Did you mean "named and designed"?

File: doc/topics/grains/index.rst (hunk 2 above)
Comment (terminalmage) [inline doc/topics/grains/index.rst]: "Most of the time execution module" ?

Maybe "Most of the time, an execution module" would be better.

File: doc/topics/grains/index.rst [hunk 4]
```diff
@@ -132,6 +132,15 @@ Using Jinja templating, only one match entry needs to be defined.
...
+or applications. Most of the time execution module called from a JINJA template
+will provide the necessary detail information for a Salt State. Consider if your
+idea for a Grain would should be developed as a execution module rather than a
```
Comment (terminalmage) [inline doc/topics/grains/index.rst]: Unnecessary capitalization.

File: doc/topics/grains/index.rst [hunk 5]
```diff
@@ -132,6 +132,15 @@ Using Jinja templating, only one match entry needs to be defined.
...
+Gain or a combination of both keeping minimum grain data.
```
Comment (terminalmage) [inline doc/topics/grains/index.rst]: Misspelling and unnecessary capitalization.

File: doc/topics/grains/index.rst [hunk 6]
```diff
@@ -132,6 +132,15 @@ Using Jinja templating, only one match entry needs to be defined.
 Writing Grains
 ==============
 
+Consider adding Gains when the information is needed for targeting,
+the grains should only contain the **bare minimum data** required. Also think
+about other platforms/operating systems, and if the grain key name and data
+structure can be named and design to support many platforms, operating systems
+or applications. Most of the time execution module called from a JINJA template
+will provide the necessary detail information for a Salt State. Consider if your
```
Comment (terminalmage) [inline doc/topics/grains/index.rst]: We try to avoid 2nd-person pronouns in our documentation where possible.

**Review by terminalmage — CHANGES_REQUESTED** [review]

**Comment by cachedout** [comment]
Thank you, @damon-atkins

**Pull request merged**